sycl/test/on-device/online_compiler/online_compiler_common.hpp

```diff
@@ -0,0 +1,181 @@
+    std::vector<byte> IL;
+    try {
+      IL = Compiler.compile(
+          std::string(CLSource),
+          // Pass two options to check that more than one is accepted.
+          std::vector<std::string>{std::string("-cl-fast-relaxed-math"),
+                                   std::string("-cl-finite-math-only")});
+      std::cout << "IL size = " << IL.size() << "\n";
+      assert(IL.size() > 0 && "Unexpected IL size");
+    } catch (sycl::exception &e) {
+      std::cout << "Compilation to IL failed: " << std::string(e.what())
+                << "\n";
+      return 1;
+    }
+#ifdef RUN_KERNELS
+    testSyclKernel(Q, getSYCLKernelWithIL(Context, IL));
+#endif // RUN_KERNELS
+  }
+
+  { // Compile and run a trivial OpenCL kernel using online_compiler()
+    // constructor accepting SYCL device.
+    std::cout << "Test case2\n";
+    online_compiler<source_language::opencl_c> Compiler(Device);
+    std::vector<byte> IL;
+    try {
+      IL = Compiler.compile(std::string(CLSource));
+      std::cout << "IL size = " << IL.size() << "\n";
+      assert(IL.size() > 0 && "Unexpected IL size");
+    } catch (sycl::exception &e) {
+      std::cout << "Compilation to IL failed: " << std::string(e.what())
+                << "\n";
+      return 1;
+    }
+#ifdef RUN_KERNELS
+    testSyclKernel(Q, getSYCLKernelWithIL(Context, IL));
+#endif // RUN_KERNELS
+  }
+
+#ifdef COMPILE_CM_KERNEL
+  // TODO: this test is temporarily turned off because CI buildbots do not set
+  // PATHs to clangFEWrapper library properly.
+  { // Compile a trivial CM kernel.
+    std::cout << "Test case3\n";
+    online_compiler<source_language::cm> Compiler;
+    try {
+      std::vector<byte> IL = Compiler.compile(std::string(CMSource));
+
+      std::cout << "IL size = " << IL.size() << "\n";
+      assert(IL.size() > 0 && "Unexpected IL size");
+    } catch (sycl::exception &e) {
+      std::cout << "Compilation to IL failed: " << std::string(e.what())
+                << "\n";
+      return 1;
+    }
+  }
+#endif // COMPILE_CM_KERNEL
+
+  { // Compile a source with syntax errors.
+    std::cout << "Test case4\n";
+    online_compiler<source_language::opencl_c> Compiler;
+    std::vector<byte> IL;
+    bool TestPassed = false;
+    try {
+      IL = Compiler.compile(std::string(CLSourceSyntaxError));
+    } catch (sycl::exception &e) {
+      std::string Msg = e.what();
+      if (Msg.find("syntax error here") != std::string::npos)
+        TestPassed = true;
+      else
+        std::cerr << "Unexpected exception: " << Msg << "\n";
+    }
+    assert(TestPassed && "Failed to throw an exception for syntax error");
+    if (!TestPassed)
+      return 1;
+  }
+
+  { // Compile a good CL source using unrecognized compilation options.
+    std::cout << "Test case5\n";
+    online_compiler<source_language::opencl_c> Compiler;
+    std::vector<byte> IL;
+    bool TestPassed = false;
+    try {
+      IL = Compiler.compile(
+          std::string(CLSource),
```

Comment:
```suggestion
          CLSource,
```

Reply:
Fixed in 893852b81738ba82be3748c1cfcbabc8f61bd94d